openhands/runtime/impl/docker/docker_runtime.py

```diff
@@ -180,22 +188,32 @@ def _init_container(self):
             plugin_arg = (
                 f'--plugins {" ".join([plugin.name for plugin in self.plugins])} '
             )
-        self._host_port = self._find_available_port()
-        self._container_port = (
-            self._host_port
-        )  # in future this might differ from host port
+        self._host_port = self._find_available_port(EXECUTION_SERVER_PORT_RANGE)
+        self._container_port = self._host_port
+        self._vscode_port = self._find_available_port(VSCODE_PORT_RANGE)
+        self._app_ports = [self._find_available_port(APP_PORT_RANGE) for _ in range(2)]
```

Comment:
Seems this would return two identical ports?

Reply:
huh...it doesn't, but now that you say it I don't know why....

Reply:
It's grabbing random ports, so there's a 1-in-10k chance of a collision. Will fix